dev/bots/analyze.dart

```diff
@@ -153,6 +158,34 @@ Future<void> run(List<String> arguments) async {
 
 // TESTS
 
+Future<void> verifyNoSyncAsyncStar(String workingDirectory, {int minimumMatches = 2000 }) async {
+  final RegExp syncPattern = RegExp(r'\s*?a?sync\*\s*?{');
+  const String ignorePattern = 'no_sync_async_star';
+  final RegExp commentPattern = RegExp(r'^\s*?///?');
+  final List<String> errors = <String>[];
+  await for (final File file in _allFiles(workingDirectory, 'dart', minimumMatches: minimumMatches)) {
+    if (file.path.contains('test')) {
+      continue;
+    }
+    final List<String> lines = file.readAsLinesSync();
+    for (int index = 0; index < lines.length; index += 1) {
+      final String line = lines[index];
+      if (line.startsWith(commentPattern)) {
+        continue;
+      }
+      if (line.contains(syncPattern) && !line.contains(ignorePattern) && (index == 0 || !lines[index - 1].contains(ignorePattern))) {
+        errors.add('${file.path}:$index: sync*/async* without an ignore (no_sync_async_star).');
+      }
+    }
+  }
+  if (errors.isNotEmpty) {
+    exitWithError(<String>[
+      '${bold}Do not use sync*/async* methods.$reset',
```

Comment:
should probably mention to see the style guide for details (and then include information as to why we feel this way in the style guide, otherwise we'll eventually forget and either (a) continue cargo-culting this even when dart is better, or (b) remove it because we want to avoid cargo culting, even though dart is not better)

Reply:
Done.